sqlalchemy_filters/filters.py

```diff
@@ -72,6 +85,39 @@ def format_for_sqlalchemy(self):
             return function(field, self.value)
 
 
+def build_sqlalchemy_filters(filterdef, models):
+    """ Recursively parse the `filterdef` into sqlalchemy filter arguments """
+
+    if isinstance(filterdef, list):
+        return [build_sqlalchemy_filters(item, models) for item in filterdef]
+
+    if isinstance(filterdef, dict):
+        # Check if filterdef defines a boolean function.
+        for key, boolean_fn, only_one_arg in BOOLEAN_FUNCTIONS:
+            if key in filterdef:
+                # The filterdef is for a boolean-function
+                # Get the function argument definitions.
+                fn_args = filterdef[key]
+                # validate the arguments
+                if not isinstance(fn_args, list):
+                    raise BadFilterFormat(
+                        '`{}` value must be a list or tuple'.format(key)
+                    )
+                if only_one_arg and len(fn_args) != 1:
+                    raise BadFilterFormat(
+                        '`{}` value must be a list of length 1'.format(key)
+                    )
+                if not only_one_arg and len(fn_args) <= 1:
```

Comment:
I wonder if we should raise with sequences containing only one argument.

Apply `or` or `and` to sequences with only one element could not make much sense, but it seems a grammatically valid option for the operator, so I think we should do `< 1` instead and allow sequences of at least one element here.